### ISSUE TITLE
[Bug Report] Unable to make websocket connection: see browser console for details

### ISSUE THREAD

**Issue by phantasy36** [stashapp/stash#4973]
**Describe the bug**

**To Reproduce**
1. Delete the 0.26.0 docker image
2. Docker compose install 0.26.1
3. Use the original 0.26.0 version docker image configuration
4. An error occurred while accessing the web page 192.168.8.60:9999

**Screenshots**
<img width="1241" alt="image" src="https://github.com/stashapp/stash/assets/46774023/637800f8-f7d5-4bb2-b2b8-4526f71a9b2b">

**Stash Version: (from Settings -> About):0.26.1

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Browser [e.g. chrome, safari] edge
 - Version [e.g. 22] 24h2

**Comment by Alextocode** [comment]
Hi there,
I upgraded from 0.25.1 to 0.26.1 and I am seeing the same issue.
I am also using Docker-Compose on an Ubuntu 22.04.3 VM.


Not sure if it is relevant but my Stash currently is proxied via Apache HTTPD 2.4 - also on (another) Ubuntu 22.04.3 VM
This is my Virtual Host config:

`ServerAdmin webmaster@localhost
ErrorLog /var/log/apache2/error.log
CustomLog /var/log/apache2/access.log combined
ServerAlias XXXXX.de
ServerName XXXXX.de

ProxyPass / http://192.168.XX.XX/
ProxyPassReverse / http://192.168.XX.XX/
RequestHeader setIfEmpty X-Forwarded-Prefix /
ProxyPreserveHost on

RewriteEngine on
RewriteCond %{HTTP:Upgrade} websocket [NC]
RewriteCond %{HTTP:Connection} upgrade [NC]
RewriteRule ^/?stash/(.*) "ws://192.168.XX.XX/$1" [P,L]

SSLEngine on
SSLProtocol +TLSv1.2 +TLSv1.3
SSLCertificateFile /etc/apache2/certs/star_XXXX_de_chained.crt
SSLCertificateKeyFile /etc/apache2/certs/star_XXXX_de.key
`

The general apache2.conf is not modified and has its defaults set.
Client side I am using Windows 10 and Firefox latest version. There must be something wrong I guess.

We see errors in console like this:
![grafik](https://github.com/stashapp/stash/assets/11820339/95004b10-7ee9-4cb8-a246-1865eeca889a)
The error happens more or less randomly while browsing in Stash.

Hope my information help to resolve it.

Let me know in case you will need further information and keep up the great work :)

Cheers!

Alex

**Comment by d3f3n** [comment]
I am using it as docker image on a synology. I have the same error after the update

**Comment by lukeh990** [comment]
I was using a NGINX reverse proxy and the reason is without specifically allowing it NGINX doesn't allow the client to upgrade to the WS protocol. Adding these lines to my config worked:
```
proxy_set_header Upgrade $http_upgrade;
proxy_set_header Connection "upgrade";
```

**Comment by Alextocode** [comment]
> I was using a NGINX reverse proxy and the reason is without specifically allowing it NGINX doesn't allow the client to upgrade to the WS protocol. Adding these lines to my config worked:
> 
> ```
> proxy_set_header Upgrade $http_upgrade;
> proxy_set_header Connection "upgrade";
> ```

Excellent reply. This was spot on for me after some try and error.
For me it got fixed to after altering my Apache HTTPD Virtual Host:

Just change:
`ProxyPass / http://192.168.XX.XX/`

to
`ProxyPass / http://192.168.XX.XX/ upgrade=websocket`

and restart Apache HTTPD. It is not necessary to activate modules like proxy_wstunnel or so.

**Comment by G2G2G2G** [comment]
Yea this isn't a bug unless it is happening without behind behind any reverse proxy.
There used to be no socket, now there is, so it requires new config for reverse proxy.
Docker maybe requires something too. but the program itself runs and does the socket fine.

**Comment by electronicron** [comment]
> I am using it as docker image on a synology. I have the same error after the update

Did you find a solution to this? I'm having the same issue.

**Comment by electronicron** [comment]
> Yea this isn't a bug unless it is happening without behind behind any reverse proxy. There used to be no socket, now there is, so it requires new config for reverse proxy. Docker maybe requires something too. but the program itself runs and does the socket fine.

I'm not sure I understand. Is there a solution to this error? I'm implementing it in a container on a synology nas and getting the same error.

Thanks!

**Comment by electronicron** [comment]
Yes, I have done this and the error persists. Could there be any other solution? Thx.

**Comment by unLieb** [comment]
> I was using a NGINX reverse proxy and the reason is without specifically allowing it NGINX doesn't allow the client to upgrade to the WS protocol. Adding these lines to my config worked:
> 
> ```
> proxy_set_header Upgrade $http_upgrade;
> proxy_set_header Connection "upgrade";
> ```

Sorry for the perhaps stupid question, but where do I have to insert this? In the Advanced settings directly in NGINX or where?

**Comment by unLieb** [comment]
It's a pity that nobody here can or wants to help.

**Comment by feederbox826** [comment]
I walked a user through the steps in discord ~1mo ago
https://mlohr.com/websockets-for-synology-dsm/

**Comment by h0m3b0y** [comment]
> I walked a user through the steps in discord ~1mo ago https://mlohr.com/websockets-for-synology-dsm/
Updated Stash today, got the error, this was a very quick fix in DSM. Thank you!
